pkg/translator/cloudwatch/cwmetrics_to_metrics.go

```diff
@@ -0,0 +1,165 @@
+package cloudwatch
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	expmetrics "github.com/open-telemetry/opentelemetry-collector-contrib/internal/exp/metrics"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	conventions "go.opentelemetry.io/collector/semconv/v1.27.0"
+	"io"
+	"strings"
+	"time"
+)
+
+// The cloudwatchMetric is the format for the CloudWatch metric stream records.
+//
+// More details can be found at:
+// https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch-metric-streams-formats-json.html
+type cloudwatchMetric struct {
+	MetricStreamName string                 `json:"metric_stream_name"`
+	AccountID        string                 `json:"account_id"`
+	Region           string                 `json:"region"`
+	Namespace        string                 `json:"namespace"`
+	MetricName       string                 `json:"metric_name"`
+	Dimensions       map[string]string      `json:"dimensions"`
+	Timestamp        int64                  `json:"timestamp"`
+	Value            *cloudwatchMetricValue `json:"value"`
+	Unit             string                 `json:"unit"`
+}
+
+// The cloudwatchMetricValue is the actual values of the CloudWatch metric.
+type cloudwatchMetricValue struct {
+	Max   float64 `json:"max"`
+	Min   float64 `json:"min"`
+	Sum   float64 `json:"sum"`
+	Count float64 `json:"count"`
+	Q1    float64 `json:"p99"`
+	Q2    float64 `json:"p99.9"`
+}
+
+const (
+	attributeAWSCloudWatchMetricStreamName = "aws.cloudwatch.metric_stream_name"
+	dimensionInstanceID                    = "InstanceId"
+	namespaceDelimiter                     = "/"
+)
+
+// isMetricValid validates that the cloudwatch metric has been unmarshalled correctly
+func isMetricValid(metric cloudwatchMetric) (bool, error) {
+	if metric.MetricName == "" {
+		return false, errors.New("cloudwatch metric is missing metric name field")
+	}
+	if metric.Namespace == "" {
+		return false, errors.New("cloudwatch metric is missing namespace field")
+	}
+	if metric.Unit == "" {
+		return false, errors.New("cloudwatch metric is missing unit field")
+	}
```

Comment:
It is my understand that a cloudwatch Unit has a fixed set of values, so it can't just accept any. 

Not strictly worth fixing now, but something to call out.

Reply:
I have added all the possible valid types. I was a bit hesitant on this, because I am afraid it will require more maintenance, but time will tell.